python/lsst/meas/extensions/piff/piffPsfDeterminer.py

```diff
@@ -585,12 +588,13 @@ def _get_threshold(nth_order):
 
         drawSize = 2*np.floor(0.5*stampSize/self.config.samplingSize) + 1
 
-        used_image_pos = [s.image_pos for s in piffResult.stars if not s.is_flagged and not s.is_reserve]
+        used_image_starId = {s.data.properties['starId'] for s in piffResult.stars
+                             if not s.is_flagged and not s.is_reserve}
```

Comment:
Let's add a guard that ensures that these IDs were unique to begin with, and that we didn't have multiple sources that were all set to have an ID of 0 or -1 or something like that. I'm thinking along the lines of

```python
assert len(used_image_starId) == nUsedStars, "Star IDs are not unique"
```